Utilities/bootstrap

```diff
@@ -804,6 +841,8 @@ def get_swiftpm_flags(args):
     cross_compile_hosts = args.cross_compile_hosts
     if build_target == 'x86_64-apple-macosx' and "macosx-arm64" in cross_compile_hosts:
```

Comment:
I realize that this is existing code, but as I read it, this will only do the right thing if I'm building for Apple Silicon on an Intel machine.  Building a universal toolchain on an Apple Silicon machine doesn't seem to be supported, since `built_target` won't be `x86_64-apple-macosx`.  Am I misreading this?

Reply:
@shahmishal Do you know?

Reply:
Yeah, as I noted in a review comment for this pull earlier, only cross-compiling from mac x86_64 to mac arm64 is currently supported. It will error here if you go the other way.